base/shared/src/main/scala/scalaz/typeclass/ApplyClass.scala

```diff
@@ -2,5 +2,5 @@ package scalaz
 package typeclass
 
 trait ApplyClass[F[_]] extends Apply[F] with FunctorClass[F] {
-  implicit final def apply: Apply[F] = this
+  implicit final def apply: Apply[F] with this.type = this
```

Comment:
I disagree with this being `implicit`, but you didn't do that.

Reply:
Lol wtf. Fixing that.